website/docs/d/kms_secret.html.markdown

```diff
@@ -1,13 +1,15 @@
 ---
 layout: "aws"
 page_title: "AWS: aws_kms_secret"
-sidebar_current: "docs-aws-datasource-kms-secret"
+sidebar_current: "docs-aws-datasource-kms-secret-x"
 description: |-
     Provides secret data encrypted with the KMS service
 ---
 
 # Data Source: aws_kms_secret
 
+!> **WARNING:** The `aws_kms_secret` data source contains behavior that cannot be supported on Terraform 0.12+ and it will either introduce breaking changes or be removed completely in the next major version (2.0.0) of the AWS provider. You can migrate existing Terraform configurations to the `aws_kms_secrets` data source, which closely resembles this data source, following instructions available in the [`aws_kms_secrets` data source documentation](/docs/providers/aws/d/kms_secrets.html).
```

Comment:
This is subjective, but I feel like we should be more direct in what we are telling users to do here. Something like:

> **WARNING:** This data source is deprecated and will be removed in a future version of the AWS provider. Use the replacement data source [`aws_kms_secrets`](/docs/providers/aws/d/kms_secrets.html) instead for new configurations.

While indeed it's not necessarily true that it'll be removed entirely, I don't think the "may introduce changes _or_ be removed" really adds anything to the main takeaway for this paragraph, which is "stop using this data source".

Reply:
I agree with @apparentlymart here that he need a more direct way to say what is specifically happening when 0.12 drops